core/store/src/trie/iterator.rs

```diff
@@ -216,6 +216,27 @@ impl<'a> TrieIterator<'a> {
         prefix
     }
 
+    /// Visits all nodes belonging to the interval [path_begin, path_end) in depth-first search
+    /// order and return TrieItem for each visited node with value stored
+    /// Used to generate split states for re-sharding
+    pub(crate) fn get_trie_items(
```

Comment:
I thought we already have a similar function? cc @Longarithm

Reply:
The difference is that here we extract only key-value pairs, and in `visit_nodes_interval` below we extract all nodes, including intermediate ones (like branches).
I guess that's intended, because when we split a trie the new trie structure may be different.
Let's state this explicitly by renaming these functions to `get_key_value_pairs_in_interval` and `get_nodes_in_interval`?

Reply:
I see. Yeah probably makes sense to emphasize that we only care about key value pairs here and not trie nodes

Reply:
Updated the comment